streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@satishd Sorry, i did not see your last comments until now. For some reason, I am not always getting notifications.  I read your explanation about avro record with null value. It makes sense. However, here we are not dealing with avro record directly, anywhere. Either in the code that we write in SAM processors/sinks or in custom UDFs, UDAFs, CPs, etc. Ultimately, user understands that an avro record is translated to a StreamlineEvent. How we define the behavior of null avro fields in StreamlineEvent is our decision. One way of doing is putting a null value. Other is not putting an entry at all for that field.  As far as your point about map contains key returning false is concerned, do you have a use case in mind where user would want it to return true? It seems that you are saying user will rely on getting all field names in schema by doing a map.keys() or something on those lines. I dont think they should need to do that from the message payload. They should already be aware of the schema and not rely on the map to get the fields in their schema.